File: Diet Plan Performance/Leetcode_1176.cc
class Solution {
public:
    int dietPlanPerformance(vector<int>& calories, int k, int lower, int upper) {
        int T=0, points=0;
        for (int i=0; i<calories.size()+1; i++){
            if (i<k) T+=calories[i];
            else{
                if (T<lower) points--;
                if (T>upper) points++;
                if (i < calories.size()) T += (calories[i] - calories[i-k]);
                else break;
            }
        }
        return points;
    }
};

File: Diet Plan Performance/Leetcode_1176.py
class Solution:
    def dietPlanPerformance(self, calories: List[int], k: int, lower: int, upper: int) -> int:
        points = 0
        T = sum(calories[:k]) 
        i = 0
        while True:
            if T>upper:
                points += 1
            elif T<lower:
                points -= 1
            if i+k < len(calories):
                T -= calories[i]
                T += calories[i+k]
            else:
                break
            i += 1
        return points
